mvcc/backend/backend.go

```diff
@@ -187,6 +190,23 @@ func (b *backend) BatchTx() BatchTx {
 
 func (b *backend) ReadTx() ReadTx { return b.readTx }
 
+// ConcurrentReadTx creates and returns a new ReadTx, which:
+// A) creates and keeps a copy of backend.readTx.txReadBuffer,
+// B) references the boltdb read Tx (and its bucket cache) of current batch interval.
+func (b *backend) ConcurrentReadTx() ReadTx {
+	b.readTx.RLock()
+	defer b.readTx.RUnlock()
+	// prevent boltdb read Tx from been rolled back until store read Tx is done.
+	b.readTx.txWg.Add(1)
+	return &concurrentReadTx{
+		buf:     b.readTx.buf.unsafeCopy(),
```

Comment:
i remember the design doc mentioned that this is lazy copy. the first read wont need to do the copy?

Reply:
Making a copy at the beginning is just a natural way of preserving the correct view [1] throughout the life cycle of a read transaction. I think lazy copy could be a nice follow up improvement, especially if copying the buffer is time or memory consuming. So far in my benchmark tests I do not feel like this is an issue.

[1] By correct view, I mean the read buffer's state at the beginning of the read transaction, combined with the view of boltdb Tx opened at the beginning of the current batch interval.

Reply:
ok. can you add a todo or something? just in case we want to do an optimization in future.

Reply:
Sounds good. I will add a TODO.